gaphor/diagram/shapes.py

```diff
@@ -323,10 +328,19 @@ def __init__(self, text=lambda: "", width=lambda: -1, style: Style = {}):
         super().__init__(text, width, style)
         self.bounding_box = Rectangle()
 
-    def draw(self, context, bounding_box):
+    def draw(
+        self, context: cairo.Context, bounding_box: Rectangle
+    ) -> Tuple[int, int, int, int]:
+        """Draw the editable text."""
         x, y, w, h = super().draw(context, bounding_box)
+        text_box = super().text_box(bounding_box)
+        cr = context.cairo
+        text_align = self.style("text-align")
+        vertical_align = self.style("vertical-align")
+        x, y = focus_box_pos(text_box, self.size(cr), text_align, vertical_align)
```

Comment:
Running `self.size(cr)` here is not efficient: it's part of the render path and will create another `Pango.Context` under the hood. If we need those values, we'd better save them in the `size()` function, and use them here.

Instead `boundingbox.width|height` should do the trick.